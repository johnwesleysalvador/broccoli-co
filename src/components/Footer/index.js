import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';

const Wrapper = styled.div`
    min-height: 70px;
    display: flex;
    width: 100%;
    justify-content: center;
    align-items: center;
    position: fixed;
    bottom: 0;

    ${({
        theme: {
            media: { phone },
        },
    }) => phone} {
        position: relative;
    }
`;

const Footer = ({ children }) => (
    <Wrapper>
        <Typography variant="caption" align="center" color="inherit">
            {children}
        </Typography>
    </Wrapper>
);

Footer.propTypes = {
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.array]).isRequired,
};

export default Footer;
