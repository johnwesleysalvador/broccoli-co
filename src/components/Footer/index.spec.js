import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import theme from 'themes/default';
import Footer from '.';

describe('<Footer />', () => {
    test('Should render Footer', () => {
        const themeContext = shallow(<ThemeProvider theme={theme} />)
            .instance()
            .getChildContext();

        const render = shallow(<Footer>Lorem Ipsum</Footer>, { context: { ...themeContext } });
        expect(render).toMatchSnapshot();
    });
});
