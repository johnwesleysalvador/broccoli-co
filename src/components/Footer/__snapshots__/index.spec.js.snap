// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should render Footer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer>
    Lorem Ipsum
</Footer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Typography)
        align="center"
        color="inherit"
        variant="caption"
>
        Lorem Ipsum
</WithStyles(Typography)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "children": "Lorem Ipsum",
        "color": "inherit",
        "variant": "caption",
      },
      "ref": null,
      "rendered": "Lorem Ipsum",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Typography)
          align="center"
          color="inherit"
          variant="caption"
>
          Lorem Ipsum
</WithStyles(Typography)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "children": "Lorem Ipsum",
          "color": "inherit",
          "variant": "caption",
        },
        "ref": null,
        "rendered": "Lorem Ipsum",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "__styled-components__": [Function],
      "__styled-components__next__": Object {
        "getTheme": [Function],
        "subscribe": [Function],
        "unsubscribe": [Function],
      },
    },
  },
}
`;
